Added previous map filter option

TheTedder opened the pull request:
Added a checkbox that allows you to filter by the map you're leaving, not the one you're entering.
Added an UnrealLoadsSettings argument to GameMemory constructor so it can access settings and determine whether to invoke the MapChangeHandler with the current map or the previous map.
Made it so that the previous map is set to whatever map you're currently on once attached to a process, instead of an empty string.

Added all the hp2 maps.

TheTedder committed:
added previous map filter

added a checkbox that allows you to filter by the map you're leaving, not the one you're entering

TheTedder committed:
added ALL the HP2 maps

'bout damn time

TheTedder committed:
bugfix

apparently this change broke everything so i reverted it

TheTedder committed:
stopped converting maps to lowercase

This could cause problems if you had two maps that were the same name save for capitalization. No reason to strip capitalization here.

TheTedder committed:
removed timer pause on loads

TheTedder committed:
made map string comparisons ignroe case

TheTedder committed:
removed irrelevant hp2 maps

Dalet reviewed (commented):
Why remove the pause on cutscene skips?

Dalet committed:
change the settings UI

Dalet committed:
Revert "removed timer pause on loads"

This reverts commit ba1802e8651e65e808461b746c9863071a8b4219.

TheTedder committed:
made UnrealLoadsComponent handle UsePrevMapChecking

changed the MapChangeEventHandler to now accept 3 args. removed the feature where GameMemory would get a reference to Settings.

TheTedder committed:
Update GameMemory.cs

fixed a bug where prevMap wasn't getting set properly

TheTedder committed:
Update HarryPotter2.cs

added check to make sure we're either on the correct map or the game has just launched. this prevents splits from accidentally starting during practice

TheTedder committed:
Update LiveSplit.UnrealLoads.dll

update the actual dll (forgot to do this earlier)

TheTedder committed:
added missing HP3 maps

TheTedder committed:
changed var name

UsePrevMap -> SplitOnLeave
mapname -> mapName

(also removed a commented-out line)

TheTedder committed:
Delete LiveSplit.UnrealLoads.dll

apparently I wasn't supposed to have this in here (oops!)

TheTedder committed:
added HP3 autostart detection

TheTedder committed:
fix

leftover old var name

Dalet committed:
change delegate parameter to pascalCase

TheTedder committed:
fix hp3 map name

TheTedder committed:
added enter and leave checkboxes for each map

and a bunch of other things too. was kind of tunnel-visioning and forgot to separate things out into their own commits...

TheTedder committed:
removed unused field

no longer necessary due to changes in how Maps are stored in the settings

TheTedder committed:
cleanup

made some code neater

TheTedder committed:
added ability to filter by map extension

fixes a bug where loading a save would trigger a split

TheTedder committed:
removed timer pausing when game closes

this behavior didn'tmake sense in an RTA speedrun format, i.e. doing a run in a single sitting with no breaks

TheTedder committed:
fix map extension filter

TheTedder committed:
streamlined logic

got rid of local function MapChangeNotify and made it inline again like it was before

TheTedder committed:
add Map.cs to project

TheTedder committed:
restored default OnDetach() behavior

games now pause the timer when they close by default again, can be overridden on an individual game basis

TheTedder committed:
bugfix

fixed issue caused by full map path not always being included in the internal representation of the map name

TheTedder committed:
added backwards compat for settings

should work as long as this update counts as a new release

Dalet committed:
clean

Dalet committed:
use > operator for version comparison

Dalet committed:
Settings: use nameof op for databinding

Dalet committed:
Settings: SplitOnEnter enabled by default

Dalet committed:
Settings: fill window w/ growing map list

Dalet committed:
clean

Dalet committed:
Settings: give a disabled look to the map DataGridView

Dalet committed:
refactor gameMemory_OnMapChange

Dalet committed:
forgot null checks when refactoring

Dalet committed:
Settings: fix version parsing
